src/app/application.cpp

```diff
@@ -173,6 +174,13 @@ Application::Application(int &argc, char **argv)
     SettingsStorage::initInstance();
     Preferences::initInstance();
 
+    const bool firstTimeUser = !Preferences::instance()->getAcceptedLegal();
+    if (!firstTimeUser) {
+        if (!upgrade())
+            throw RuntimeError(u"Failed migration of old settings"_qs); // Not translatable. Translation isn't configured yet.
+        handleChangedDefaults(!firstTimeUser ? DefaultPreferencesMode::Legacy : DefaultPreferencesMode::Current);
```

Comment:
It should be moved away from `if (!firstTimeUser)`. Otherwise the logic is broken.

Reply:
Dumb mistake. Done. I preferred the code from main.cpp now. To avoid doubly checking the bool state.